### PR TITLE
 Create Modal Component

### PR DESCRIPTION
**Pull request by acostanzo**
Screen Shot 2019-03-14 at 12 12 45 PM
[#164384241]

* Add react-modal package
* Add story for modal
* Clear out default react-modal styles by supplying class names for the
overlay and modal content
* Style the modal and overlay via emotion css
* Allow Pane width to be overwritten by prop
* Add gridTemplateRows to Grid
* Add maxHeight to Box

### CODE REVIEW

File: src/Modal/Modal.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  bottom: 0;
+  display: flex;
+  justify-content: center;
+  left: 0;
+  position: fixed;
+  right: 0;
+  top: 0;
+`;
+
+const baseModalCss = theme => css`
```
Comment (dgalarza) [inline src/Modal/Modal.js]: What do you think about pulling this and `baseOverlayCss` out into a `styles` file within `Modal?

File: src/Modal/Modal.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  );
+};
+
+Modal.propTypes = {
+  children: PropTypes.oneOfType([
+    PropTypes.node,
+    PropTypes.arrayOf(PropTypes.node)
+  ]).isRequired,
+  modalCss: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
+  overlayCss: PropTypes.oneOfType([PropTypes.string, PropTypes.func])
```
Comment (dgalarza) [inline src/Modal/Modal.js]: I feel like when we were pairing we ran into something where these CSS props weren’t ever actually strings. Is that right?

File: src/Modal/Modal.js (hunk 1 above)
Comment (acostanzo) [reply src/Modal/Modal.js]: I'd be down with that. Clean it up a bit.

File: src/Modal/Modal.js (hunk 2 above)
Comment (acostanzo) [reply src/Modal/Modal.js]: mmm, good Q. Ill have to go back in and look

File: stories/modal.stories.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+// For modal accessibility. See http://reactcommunity.org/react-modal/accessibility/ for details
+ReactModal.setAppElement(document.body); // eslint-disable-line no-undef
+
+class ModalStory extends React.Component {
+  state = {
+    modalIsOpen: true
+  };
+
+  closeModal = () => {
```
Comment (mikecx) [inline stories/modal.stories.js]: Any thoughts on if it's more appropriate for the modal to manage its state or the parent to manage its state?

Comment (euanlau) [reply stories/modal.stories.js]: My vote is to leave it as controlled (changing modal state via prop).
If we want an uncontrolled one down the road, we could always create a wrapper.

Comment (acostanzo) [reply stories/modal.stories.js]: Whoops sorry @mikecx saw the approval before I saw your Q. I did think about that a bit. react-modal handles the dispaly of the modal via an isOpen prop. I started playing with the idea of wrapping it with our own modal component that managed state but in most (all?) cases you would be opening the modal through some interaction or event. The only way I could see it managing its own state is if it was open by default, then you could close it. However in that situation you wouldn't be able to open it again. Did you have a scenario in mind?

**Pull request merged**